docs/development/deployment/chain-spec.md

```diff
@@ -2,24 +2,150 @@
 title: Chain Specification
 ---
 
-* What is a chain specification?
+A chain specification, or "chain spec", is a collection of configuration information that dictates
+which network a blockchain node will connect to, which entities it will initially communicate with,
+and what consensus-critical state it must have at genesis.
 
-* How do I launch a chain with a custom specification?
+## Structure of a Chain Spec
 
-* What does it mean to launch a substrate chain with a different chain spec?
+The [`ChainSpec` struct](https://substrate.dev/rustdocs/master/sc_service/struct.ChainSpec.html)
+separates the information contained in a chain spec into two parts. A node can use a `ChainSpec`
+instance to create a genesis block.
 
-* How do I generate one?
+### The Spec
 
-* What can be configured/changed?
+The first part of the chain spec, the unfortunately-named `spec`, contains information the node will
+use to communicate with other parties in the network such as a set of bootnodes, a set of telemetry
+endpoints to which the node will send data, and human- and machine-readable names for the network to
+which the node will connect. Many of these items can be overridden by command-line flags, and the
+values can be changed after the blockchain has been launched.
 
-* Popular configurations
+> Caution: While all properties in this section can be changed after genesis, nodes will only add
+> peers who use the same `protocolId`.
 
-	* Initial Authorities
+### Extension
 
-	* Initial Balances
+Because the Substrate framework is extensible, it provides a way to customize the chain spec with additional data to configure parts of the node outside of the runtime. One example use case is telling the node about well-known blocks at specific heights, to prevent long range attacks when syncing a new node from genesis.
```

Comment:
The extensions are going to be mostly usable for people that want to extend `node` functionality. I.e. it's mostly usable in `substrate-client`, not necessarily in custom runtimes.

Reply:
Yes, that was my understanding from reading the code. I tried to capture that with:

> it provides a way to customize the chain spec with additional data to configure parts of the node **outside of the runtime**.

Does that seem correct? I will also explicitly mention that it is for the client.

Reply:
Yup, that sounds good!